mlrun/datastore/google_cloud_storage.py

```diff
@@ -133,12 +139,12 @@ def listdir(self, key):
 
     def rm(self, path, recursive=False, maxdepth=None):
         path = self._make_path(path)
-        self.filesystem.exists(path)
-        self.filesystem.rm(path=path, recursive=recursive, maxdepth=maxdepth)
+        if self.filesystem.exists(path):
+            self.filesystem.rm(path=path, recursive=recursive, maxdepth=maxdepth)
```

Comment:
Do we really need to check if it exists? Can't we just try to delete it?

Reply:
it will raise an error if file missing, another issue of 2024.3.1